internal/oc/release.go

```diff
@@ -442,21 +444,162 @@ func execute(log logrus.FieldLogger, executer executer.Executer, pullSecret stri
 	}
 }
 
-// Create a temporary file containing the ImageContentPolicySources
-func (r *release) getIcspFileFromRegistriesConfig(log logrus.FieldLogger) (string, error) {
+// whiteSpaceRE is the regular expression used to split commands into arguments, taking into account that the separator
+// can be one or multiple whilte spaces.
+var whiteSpaceRE = regexp.MustCompile(`\s+`)
 
+// getMirrorsFromRegistriesConfig returns the information needed to generate the command line flag that specifies the
+// mirrors configuration used by the 'oc' command.
+func (r *release) getMirrorsFlagFromRegistriesConfig(log logrus.FieldLogger) (result *mirrorsFlagInfo, err error) {
 	if !r.mirrorRegistriesBuilder.IsMirrorRegistriesConfigured() {
-		log.Debugf("No mirrors configured to build ICSP file")
+		log.Debugf("No mirrors configured to build mirrors file")
+		return
+	}
+
+	mirrorsRegistriesConfig, err := r.mirrorRegistriesBuilder.ExtractLocationMirrorDataFromRegistries()
+	if err != nil {
+		log.WithError(err).Errorf("Failed to get the mirror registries needed for mirrors file")
+		return
+	}
+
+	// Check if we should use the deprecated ImageContentPolicySource object or the new ImageDigestMirrorSet. To do
+	// so we check if the 'oc adm release info' command supports the '--idms-file' flag.
+	suportsIdmsFileFlag, err := r.supportsIdmsFileFlag(log, templateReleaseInfo)
+	if err != nil {
+		return
+	}
+	var (
+		name string
+		file string
+	)
+	if suportsIdmsFileFlag {
+		name = idmsFileFlagName
+		file, err = r.getIdmsFileFromRegistriesConfig(log, mirrorsRegistriesConfig)
+	} else {
+		name = icspFileFlagName
+		file, err = r.getIcspFileFromRegistriesConfig(log, mirrorsRegistriesConfig)
+	}
+	if err != nil {
+		return
+	}
+	result = &mirrorsFlagInfo{
+		logger: log,
+		name:   name,
+		file:   file,
+	}
+	return
+}
+
+// mirrorsFlagInfo contains the information needed to populate the '--idms-file' or '--icsp-file' command line flag that
+// is used by the 'oc' command to specify the mirrors configuration.
+type mirrorsFlagInfo struct {
+	// logger is the logger used by the methods of this type.
+	logger logrus.FieldLogger
+
+	// name is the name of the command line flag, 'idms-file' or 'icsp-file'.
+	name string
+
+	// file is the name of a temporary file containing the serialized ImageDigestMirrorSet or
+	// ImageContentSourcePolicy.
+	file string
+}
+
+// String returns the complete text of the flag, something like '--idms-file=/tmp/my-idms.yaml'.
+func (i *mirrorsFlagInfo) String() string {
+	if i == nil || i.name == "" || i.file == "" {
```

Comment:
@CrystalChun here ^ is the additional check you suggested.

Reply:
Looks great!